dbms/src/DataTypes/FieldToDataType.h

```diff
@@ -40,7 +40,7 @@ class FieldToDataType : public StaticVisitor<DataTypePtr>
     template <typename T>
     DataTypePtr operator()(const DecimalField<T> & x) const
     {
-        PrecType prec = maxDecimalPrecision<T>();
+        PrecType prec = x.getPrec();
```

Comment:
why?

Reply:
Because `maxDecimalPrecision<T>` only return 4 precision: `9/18/38/65`, but here we actaully want to have exact precision of the DecimalField.